docs/source/optim.rst

```diff
@@ -303,6 +314,172 @@ algorithms.
     lr_scheduler.OneCycleLR
     lr_scheduler.CosineAnnealingWarmRestarts
 
+How to utilize named parameters to load optimizer state dict
+--------------------------------------------
+
+The function :func:`~Optimizer.load_state_dict` store the optional ``param_names``content in the
+loaded state dict if present. However, the process of loading the optimizer state is not affected,
+as the order of the parameters matters to maintain compatibility (in case of different ordering).
+To utilize the loaded parameters names from the loaded state dict, a custom ``register_load_state_dict_pre_hook``
+needs to be implemented according to the desired behavior.
```

Comment:
Yes, load_state_dict (named_state_dict) will not base any decisions on the names (also for BC !)
All changes from the predicted (existing) behavior should be manually set by adding a "hook" to the process.

** This is beneficial since this feature is usually needed when there is a mismatch in the state dict (such as different layers, different order, etc.) and some manual adaptations are required according to the needs

Reply:
oop yea i...forgot to come back to this comment once i had read the rest of the code.